packages/graph/hash_graph/lib/graph/src/store/postgres/query/link_type.rs

```diff
@@ -0,0 +1,130 @@
+use postgres_types::ToSql;
+use type_system::LinkType;
+
+use crate::{
+    ontology::LinkTypeQueryPath,
+    store::postgres::query::{ColumnAccess, Field, Path, PostgresQueryRecord, Table, TableName},
+};
+
+impl<'q> PostgresQueryRecord<'q> for LinkType {
+    type Field = LinkTypeQueryField;
+
+    fn base_table() -> Table {
+        Table {
+            name: TableName::LinkTypes,
+            alias: None,
+        }
+    }
+
+    fn default_fields() -> &'q [Self::Field] {
+        &[LinkTypeQueryField::Schema, LinkTypeQueryField::OwnedById]
+    }
+}
+
+/// A [`Field`] available in [`LinkType`]s.
+///
+/// [`LinkType`]: type_system::LinkType
+#[derive(Debug, PartialEq, Eq)]
+pub enum LinkTypeQueryField {
+    BaseUri,
+    Version,
+    VersionId,
+    OwnedById,
+    Schema,
+    VersionedUri,
+    Title,
+    Description,
+    RelatedKeywords,
+}
+
+impl Field for LinkTypeQueryField {
+    fn table_name(&self) -> TableName {
+        match self {
+            Self::BaseUri | Self::Version => TableName::TypeIds,
+            Self::VersionId
+            | Self::OwnedById
+            | Self::Schema
+            | Self::VersionedUri
+            | Self::Title
+            | Self::Description
+            | Self::RelatedKeywords => TableName::LinkTypes,
+        }
+    }
+
+    fn column_access(&self) -> ColumnAccess {
+        match self {
+            Self::BaseUri => ColumnAccess::Table { column: "base_uri" },
+            Self::Version => ColumnAccess::Table { column: "version" },
+            Self::VersionId => ColumnAccess::Table {
+                column: "version_id",
+            },
+            Self::OwnedById => ColumnAccess::Table {
+                column: "owned_by_id",
+            },
+            Self::Schema => ColumnAccess::Table { column: "schema" },
+            Self::VersionedUri => ColumnAccess::Json {
+                column: "schema",
+                field: "$id",
+            },
+            Self::Title => ColumnAccess::Json {
+                column: "schema",
+                field: "title",
+            },
+            Self::Description => ColumnAccess::Json {
+                column: "schema",
+                field: "description",
+            },
+            Self::RelatedKeywords => ColumnAccess::Json {
+                column: "schema",
+                field: "relatedKeywords",
```

Comment:
Oh, interesting. Haven't thought about this before, but are we going to support operations other than `->>` in json? For example `@>` for "contains" would be useful for things that can be an array, such as `relatedKeywords`, `required` and `requiredLinks`.

Reply:
Yes, this is my plan for the future, but for now, only the nullability check can be used for these.